packages/SwingSet/src/hostStorage.js

```diff
@@ -1,51 +1,85 @@
 import harden from '@agoric/harden';
 
-// The "Storage API" a set of functions { has, getKeys, get, set, delete },
-// which work on string keys and accept string values. A lot of kernel-side
-// code expects to get an object which implements the Storage API, which is
-// usually associated with a write-back buffer wrapper.
+/*
+The "Storage API" is a set of functions { has, getKeys, get, set, delete } that
+work on string keys and accept string values.  A lot of kernel-side code
+expects to get an object which implements the Storage API, which is usually
+associated with a write-back buffer wrapper.
 
-// The "HostDB API" is a different set of functions { has, getKeys, get,
-// applyBatch } which the host is expected to provide to the Controller in
-// the config object. This API allows SwingSet to deliver batches of changes
-// to the host-side storage medium.
+The "HostDB API" is a different set of functions { has, getKeys, get,
+applyBatch } which the host is expected to provide to the Controller in the
+config object. This API allows SwingSet to deliver batches of changes to the
+host-side storage medium.
 
-// buildStorageInMemory() returns a RAM-based StorageAPI object, which can
-// accept an optional initialState string, and which provides an extra
-// getState() function to extract a string with all the state.
+buildHostDBInMemory creates hostDB objects for testing and casual hosts that
+can afford to hold all state in RAM. They must arrange to call getState() at
+the end of each block and save the resulting string to disk.
 
-// buildHostDBInMemory creates hostDB objects for testing and casual hosts
-// that can afford to hold all state in RAM. They must arrange to call
-// getState() at the end of each block and save the resulting string to
-// disk.
-
-// A more sophisticated host will build a hostDB that writes changes to disk
-// directly.
+A more sophisticated host will build a hostDB that writes changes to disk
+directly.
+*/
 
+/**
+ * Create a new instance of a RAM-based implementation of the Storage API.
+ *
+ * In addition to the regular storage API, it provides an extra getState()
+ * function that returns all the currently stored state as an object.
+ *
+ * @param initialState  An optional initial state object whose properties are
+ *    mapped to state key/value pairs.
+ *
+ * @return an object: {
+ *   storage,  // the storage API object itself
+ *   getState, // a function that returns the current state as an object
+ *   map       // the underlying map that holds the state in memory
+ * }
+ *
+ * XXX I presume providing the underlying map is just a debug thing?  Should
```

Comment:
Hmmm.  Digging more deeply, it appears to be used a couple of places in cosmic-swingset

Reply:
Looking more closely at cosmic-swingset, the code that uses the map directly seems to be doing so primarily for performance reasons, since it could indirectly use the getKeys call to achieve the same result.  Said code is iterating the store to persist it, which will go away once the state is kept in a persistent medium directly.  Given that consideration, it probably makes most sense to leave the API as is for now, with an open issue to clean this up after persistent state is implemented.

Reply:
Filed as issue #460